### PR TITLE
added current napari dependency pins

### LINKED ISSUE #7: no dependency pins in conda-package
it looks like the conda package for napari does not contain any dependency pinning (except Python). Here the `info.json` from the current (0.33) package:
<details>

```json
  "depends": [
    "backcall",
    "cachey",
    "dask",
    "fsspec",
    "imageio",
    "ipykernel",
    "ipython",
    "jupyter_client",
    "napari-plugin-engine",
    "napari-svg",
    "numpy",
    "numpydoc",
    "pluggy",
    "pyopengl",
    "pyside2",
    "python >=3.6",
    "qtconsole",
    "qtpy",
    "scikit-image",
    "scipy",
    "tifffile",
    "vispy",
    "wrapt",
    "zarr"
  ],
```
</details>

napari itself seems to have quite a few pins:

<details>

from `setup.cfg`:

```ini
install_requires =
    cachey>=0.2.1
    dask[array]>=2.1.0
    imageio>=2.5.0
    importlib-metadata>=1.5.0 ; python_version < '3.8'
    ipykernel>=5.1.1
    IPython>=7.7.0
    napari-plugin-engine>=0.1.5
    napari-svg>=0.1.3
    numpy>=1.10.0
    numpydoc>=0.9.2
    Pillow!=7.1.0,!=7.1.1  # not a direct dependency, but 7.1.0 and 7.1.1 broke imageio
    psutil>=5.0
    PyOpenGL>=3.1.0
    PyYAML>=5.1
    qtconsole>=4.5.1
    qtpy>=1.7.0
    scipy>=1.2.0
    tifffile>=2020.2.16
    toolz>=0.10.0
    vispy>=0.6.4
    wrapt>=1.11.1
```

</details>

Should these maybe be added to the recipe here?

### PR DESCRIPTION
**Pull request by k-dominik**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

closes #7 

Added current napari dependency pins. Needs https://github.com/conda-forge/napari-svg-feedstock/pull/2 (and 0.1.3 to be built)

Personally I would probably order the dependencies alphabetically...

### CODE REVIEW

**Comment by k-dominik** [comment]
@conda-forge-admin, please rerender

**Comment by jni** [comment]
@k-dominik do you mind pushing an empty commit to trigger CI again? This was missing `napari-svg` 0.1.3 from conda-forge, which didn't exist yet. This might be one of the issues that @jakirkham was trying to help me avoid when he said don't use pins. Again, I'd love to hear from him before adding all the pins to both repos. Some pins are important, others just happened to be the version we were using at the time...

**Comment by jni** [comment]
@k-dominik another thing, if you want a quick merge, then I suggest opening a separate PR that fixes the vispy issue (since I know for sure that we need 0.6.4, so that's an easy pin to add). For most of the others, I would like to examine each as well as get @jakirkham's advice before taking the leap.

**Comment by k-dominik** [comment]
> @k-dominik another thing, if you want a quick merge, then I suggest opening a separate PR that fixes the vispy issue (since I know for sure that we need 0.6.4, so that's an easy pin to add). For most of the others, I would like to examine each as well as get @jakirkham's advice before taking the leap.

sounds great, thank you very much for being so responsive!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,36 +23,46 @@ requirements:
...
+    - qtpy >=1.7.0
+    - qtconsole >=4.5.1
     - IPython
     - backcall
+    - Pillow !=7.1.0,!=7.1.1  # not a direct dependency, but 7.1.0 and 7.1.1 broke imageio
     - pluggy
-    - PyOpenGl
+    - psutil >=5.0
+    - PyOpenGL >=3.1.0
     - PySide2
```
Comment (jakirkham) [inline recipe/meta.yaml:46]: Would lower case this as well (even though it precedes this PR). Might as well get all of this updated :)

```suggestion
    - pyside2
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,36 +23,46 @@ requirements:
...
-    - wrapt
-    - numpydoc
-    - napari-plugin-engine
-    - napari-svg
-    - tifffile
+    - PyYAML >=5.1
```
Comment (jakirkham) [inline recipe/meta.yaml:47]:
```suggestion
    - pyyaml >=5.1
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,36 +23,46 @@ requirements:
   run:
     - python >=3.6
     - numpy
```
Comment (jakirkham) [inline recipe/meta.yaml:25]: Would drop this line since it was added below.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -23,36 +23,46 @@ requirements:
...
-    - napari-svg
-    - tifffile
+    - PyYAML >=5.1
+    - wrapt >=1.11.1
+    - numpy >=1.10.0
+    - numpydoc >=0.9.2
+    - napari-plugin-engine >=0.1.5
+    - napari-svg >=0.1.3
+    - tifffile >=2020.2.16
+    - toolz >=0.10.0
```
Comment (jakirkham) [inline recipe/meta.yaml:54]: Does `napari` need `toolz`? Asking as this appears to not have been listed previously.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -23,36 +23,46 @@ requirements:
...
+    - imageio >=2.5.0
+    - importlib-metadata >=1.5.0
+    - ipykernel >=5.1.1
+    - IPython >=7.7.0
     - jupyter_client
-    - qtpy
-    - qtconsole
+    - qtpy >=1.7.0
+    - qtconsole >=4.5.1
     - IPython
```
[the remark this answers is not in the document]
Comment (k-dominik) [reply recipe/meta.yaml:40]: Hi @jakirkham , thank you very much for the review, I don't know how I have missed it so long. I have to admit that I don't fully understand this suggestion.

File: recipe/meta.yaml (hunk 4 above)
Comment (k-dominik) [reply recipe/meta.yaml:57]: I think it does, see [here](https://github.com/napari/napari/blob/ab741703a2e0457abf69337bb7b15a39b2a39216/napari/layers/transforms.py#L1) and [here](https://github.com/napari/napari/blob/af011f9a78a13d46f0d67bd0c7106504243683e9/napari/_qt/threading.py#L6)

**Comment by k-dominik** [comment]
this is currently failing because of https://github.com/conda-forge/cachey-feedstock/pull/1

I don't think I can trigger a rebuild there, but building cachey locally worked.

**Comment by jakirkham** [comment]
FWIW we have [a bot command]( https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-restart-ci ) to restart CI. There are other bot commands listed in the docs as well.

**Comment by k-dominik** [comment]
> FWIW we have [a bot command](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-restart-ci) to restart CI. There are other bot commands listed in the docs as well.

It's a very busy morning, sorry. I thought that there must be something but that was quicker for me. Thank you for the link! I'll make sure to use the bots more in the future.

**Comment by jakirkham** [comment]
No need to be sorry. Pushing an empty commit is also fine. Just trying to share info 🙂

**Comment by jni** [comment]
@jakirkham I think you told me at the EOSS meeting that it's better in general not to put too many version constraints in... And if my memory is not fooling me, how do you feel about this PR?

**Comment by jni** [comment]
Also, on the PyPI side we now have a base package that doesn't depend on either pyqt or pyside and options for those. We should probably do the same here. The [matplotlib recipe](https://github.com/conda-forge/matplotlib-feedstock/blob/master/recipe/meta.yaml) is informative for this.

**Comment by jni** [comment]
@k-dominik I included your commits in #13, which is finally in! :tada: thank you for this!

**Pull request closed without merging**